### PR TITLE
GH-34214: [C++] Pass OPENSSL_ROOT_HINT to CMAKE_PREFIX_PATH for bundled AWS

### LINKED ISSUE #34214: [C++] Bundled AWS::c_cal can't find libcrypto on manylinux
### Describe the bug, including details regarding any error messages, version, and platform.

Build failures: https://github.com/apache/arrow/pull/34159#issuecomment-1432012081. We should pass OPENSSL_ROOT_HINT to CMAKE_PREFIX_PATH for bundled AWS.

### Component(s)

C++

### PR DESCRIPTION
**Pull request by js8544**
To fix build failures such as https://github.com/apache/arrow/pull/34159#issuecomment-1432012081, we need to help it find OpenSSL.
* Closes: #34214

### CODE REVIEW

**Comment by js8544** [comment]
@github-actions crossbow submit -g wheel

**Comment by js8544** [comment]
@github-actions crossbow submit -g wheel

**Comment by js8544** [comment]
@github-actions crossbow submit -g wheel -g cpp -g r java-jars

**Comment by js8544** [comment]
@github-actions crossbow submit -g wheel -g cpp -g r java-jars

**Comment by js8544** [comment]
@github-actions crossbow submit -g wheel -g cpp -g r java-jars

**Comment by js8544** [comment]
@github-actions crossbow submit -g wheel -g cpp -g r java-jars

**Comment by js8544** [comment]
@github-actions crossbow submit -g wheel -g cpp -g r java-jars

**Comment by js8544** [comment]
@kou manylinux builds are good now :)

File: cpp/cmake_modules/ThirdpartyToolchain.cmake [hunk 1]
```diff
@@ -4745,9 +4745,8 @@ macro(build_awssdk)
       -DENABLE_TESTING=OFF
       -DENABLE_UNITY_BUILD=ON
       -DOPENSSL_ROOT_DIR=${OPENSSL_ROOT_HINT}
-      "-DCMAKE_INSTALL_PREFIX=${AWSSDK_PREFIX}"
-      "-DCMAKE_PREFIX_PATH=${AWSSDK_PREFIX}")
-
+      -DCMAKE_INSTALL_PREFIX=${AWSSDK_PREFIX}
+      -DCMAKE_PREFIX_PATH=${AWSSDK_PREFIX}\\$<SEMICOLON>${OPENSSL_ROOT_HINT})
```
Comment (kou) [inline cpp/cmake_modules/ThirdpartyToolchain.cmake:4749]: Could you sort this list?

**Review by kou — APPROVED** [review]
+1

Thanks! Could you fix order before we merge this?

File: cpp/cmake_modules/ThirdpartyToolchain.cmake (hunk 1 above)
Comment (js8544) [reply cpp/cmake_modules/ThirdpartyToolchain.cmake]: Like this? https://github.com/apache/arrow/pull/34215/commits/455e60ad1c973b300557f810cc13eef3092a757c

Comment (kou) [reply cpp/cmake_modules/ThirdpartyToolchain.cmake]: Yes. :-)

**Pull request merged**